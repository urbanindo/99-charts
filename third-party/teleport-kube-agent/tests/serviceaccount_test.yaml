suite: ServiceAccount
templates:
  - serviceaccount.yaml
tests:
  - it: sets ServiceAccount labels when specified
    values:
      - ../.lint/extra-labels.yaml
    asserts:
      - equal:
          path: metadata.labels.app\.kubernetes\.io/name
          value: teleport-kube-agent
      - equal:
          path: metadata.labels.resource
          value: serviceaccount
      - matchSnapshot: {}

  - it: sets ServiceAccount annotations when specified
    values:
      - ../.lint/annotations.yaml
    asserts:
      - equal:
          path: metadata.annotations.kubernetes\.io/serviceaccount
          value: test-annotation
      - equal:
          path: metadata.annotations.kubernetes\.io/serviceaccount-different
          value: 5
      - matchSnapshot: {}
  - it: skips ServiceAccount creation
    set:
      serviceAccount.create: false
    asserts:
      - hasDocuments:
          count: 0