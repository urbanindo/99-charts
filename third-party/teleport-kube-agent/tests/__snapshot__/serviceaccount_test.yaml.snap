sets ServiceAccount annotations when specified:
  1: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      annotations:
        kubernetes.io/serviceaccount: test-annotation
        kubernetes.io/serviceaccount-different: 5
      name: RELEASE-NAME
      namespace: NAMESPACE
sets ServiceAccount labels when specified:
  1: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      labels:
        app.kubernetes.io/name: teleport-kube-agent
        resource: serviceaccount
      name: RELEASE-NAME
      namespace: NAMESPACE
